src/nft.vy

```diff
@@ -567,6 +573,23 @@ def safe_mint(owner: address, uri: String[432]):
     self._safe_mint(owner, token_id, b"")
     self._set_token_uri(token_id, uri)
 
+@external
+def mint_and_start_auction(uri: String[432], patron: address):
+    """
+    @notice Only authorized minters can access this function.
+    @param uri The maximum 432-character user-readable string URI for computing `tokenURI`.
```

Comment:
Is 432 character necessary? make sure this number is as small as possible without causing issue storing URI, this can also opt-out base-uri which is common

Reply:
ack: reduced to 176+80. can finetune later

nack: opt-out base uri seems less common to me, the current setup should be good.

Reply:
We don't need to waste storage space by storing redundant common base string, checkout logic of tokenUri function in ERC721 standard, it concats the base uri with stored token uri.

Reply:
https://github.com/SaturnCassini/auction_house_contracts/blob/ef5cadf454eab371ab03ea20df6b5d61b652b12b/src/nft.vy#L457-L484

Reply:
tbd